File: hash/java/src/main/java/io/rosetta/imagehash/Hex.java
package io.rosetta.imagehash;

import io.rosetta.imagehash.internal.BitPack;

public final class Hex {
    public static ImageHash hexToHash(String hex) {
        boolean[][] bits = BitPack.unpackSquare(hex);
        return new ImageHash(bits);
    }

    public static ImageHash hexToFlathash(String hex, int hashSize) {
        if (hashSize < 1) throw new IllegalArgumentException("hashSize must be >= 1");
        boolean[][] bits = BitPack.unpackFlat(hex, hashSize);
        return new ImageHash(bits);
    }

    private Hex() {}
}

File: hash/go/imagehash/hex.go
package imagehash

import (
	"fmt"

	"github.com/wmetcalf/rosetta-image-hash/go/imagehash/internal/bitpack"
)

func HexToHash(hex string) (Hash, error) {
	bits, err := bitpack.UnpackSquare(hex)
	if err != nil {
		return Hash{}, err
	}
	return newHashFromBits(bits), nil
}

func HexToFlathash(hex string, hashSize int) (Hash, error) {
	if hashSize < 1 {
		return Hash{}, fmt.Errorf("hashSize must be >= 1")
	}
	bits, err := bitpack.UnpackFlat(hex, hashSize)
	if err != nil {
		return Hash{}, err
	}
	return newHashFromBits(bits), nil
}
